enterprise/ssl-impl/src/main/java/io/crate/protocols/ssl/FileWatcher.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.protocols.ssl;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+class FileWatcher {
```

Comment:
This abstraction here looks a bit odd. It has no encapsulation as it exposes everything and all the methods are rather shallow.

I wonder if it would be simpler if `FileWatcher` and `FilesWatcher` would be merged.

Maybe so that you've an interface like this:

```
FilesWatcher.addListener(Path, WatchEvent.Kind, Consumer)
```

All this code here would then become private internals of the `FilesWatcher`

Reply:
i'll merge it, had it merged in the first implementation.